### PR TITLE
Supporting indexing EachReplaceMissing and EachFailMissing

### PR DESCRIPTION
**Pull request by nalimilan**
Similar to `SkipMissing`.

(I need this to support missing values at https://github.com/JuliaData/DataFrames.jl/pull/2610.)

### CODE REVIEW

**Comment by nalimilan** [comment]
Failure on nightly is https://github.com/JuliaData/DataAPI.jl/pull/32.

File: src/Missings.jl [hunk 1]
```diff
@@ -87,6 +92,8 @@ Base.IteratorEltype(::Type{<:EachReplaceMissing{T}}) where {T} =
 Base.length(itr::EachReplaceMissing) = length(itr.x)
 Base.size(itr::EachReplaceMissing) = size(itr.x)
 Base.eltype(itr::EachReplaceMissing) = nonmissingtype(eltype(itr.x))
+Base.eachindex(itr::EachReplaceMissing) = eachindex(itr.x)
+Base.keys(itr::EachReplaceMissing) = keys(itr.x)
```
Comment (bkamins) [inline src/Missings.jl:96]: These two will fail if `itr.x` does not define them, but maybe it is OK.

File: src/Missings.jl [hunk 2]
```diff
@@ -102,11 +109,19 @@ end
     return (v isa Missing ? itr.replacement : v, s)
 end
 
+Base.@propagate_inbounds function Base.getindex(itr::EachReplaceMissing, I...)
```
Comment (bkamins) [inline src/Missings.jl:112]: same here.

**Review by bkamins — APPROVED** [review]

**Comment by bkamins** [comment]
Just please check the failure of CI on nightly.

File: src/Missings.jl (hunk 1 above)
Comment (nalimilan) [reply src/Missings.jl:96]: Yes that's intended. We do the same for `SkipMissing` in Base.

Comment (bkamins) [reply src/Missings.jl:96]: so maybe we should support `length`, `size` etc. (all the array interface) in the same way? We cannot do it for `SkipMissing` but here we can.

Comment (nalimilan) [reply src/Missings.jl:96]: We already do. :-)

Comment (nalimilan) [reply src/Missings.jl:96]: Ah, we need `axes`. Will add it.

**Comment by nalimilan** [comment]
CI failure on nightly is fixed by https://github.com/JuliaData/Missings.jl/pull/127.

**Comment by nalimilan** [comment]
https://github.com/JuliaRegistries/General/pull/28600

**Pull request merged**